src/common/settingsProvider.ts

```diff
@@ -47,7 +48,10 @@ export class SettingsProvider {
   getThemeFromUserSetting(): string {
       const themeSetting = vscode.workspace.getConfiguration().get('workbench.colorTheme');
       const legacySetting = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('themes');
-      return SUPPORTED_THEMES.get((themeSetting || legacySetting) as string) || 'systemPreference';
+      if (SUPPORTED_THEMES.has(themeSetting as string)) {
+        return SUPPORTED_THEMES.get(themeSetting as string) || 'systemPreferred';
+      }
+      return SUPPORTED_THEMES.get(legacySetting as string) || 'systemPreferred';
```

Comment:
IMO doing the type casting in the declaration improved readability, what do you think of something like this:
```suggestion
      const themeSetting = vscode.workspace.getConfiguration().get('workbench.colorTheme') as string;
      const legacySetting = (vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('themes') || '') as string;
      return SUPPORTED_THEMES.get(themeSetting) || SUPPORTED_THEMES.get(legacySetting) || 'systemPreferred';
```